test/rpc.spec.js

```diff
@@ -0,0 +1,59 @@
+'use strict'
+
+const expect = require('expect')
+const rewire = require('rewire')
+
+let Rpc = rewire('../server/lib/rpc')
+
+describe('[Rpc]', () => {
+  describe('(getInfo)', () => {
+    beforeEach(() => { // reset the rewired functions
+      Rpc = rewire('../server/lib/rpc')
+    })
+    it('should run getinfo and fail the rpc call', (done) => {
+      Rpc.navClient = {
+        getInfo: () => { return Promise.reject({ code: -17 }) },
+      }
+      const res = {
+        send: (response) => {
+          const jsonResponse = JSON.parse(response)
+          expect(jsonResponse.type).toBe('FAIL')
+          expect(jsonResponse.code).toBe('RPC_002')
+          expect(jsonResponse.error.code).toBe(-17)
```

Comment:
We might also want to check for the existence of a message if throwing a message is standard expected behaviour for these try-catches